### PR TITLE
Issue 739 by pjohans: allow currencies other than Kr.

### PR DESCRIPTION
**Pull request by pjohans**

### CODE REVIEW

**Comment by pjohans** [comment]
This  one works in conjunction with:
https://github.com/ding2/ddbasic/pull/8

File: ding_event.admin.inc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * @param $form_state
+ */
+function ding_event_admin_currency_validate($form, &$form_state){
+  $currency = $form_state['values']['ding_event_currency_type'];
+  if(module_exists('ding_place2book') && $currency != 'kr'){
+    drupal_set_message(t('place2book does not support currencies other than Kr. disable ding_place2book module to continue'),'warning');
+    $form_state['input']['ding_event_currency_type'] = 'kr';
+    $form_state['rebuild'] = TRUE;
+    return;
+  }
```
Comment (kasperg) [inline ding_event.admin.inc]: I do not think this is the right way to go about this. If this is a problem for `ding_place2book` then that module should add a validate callback to the form provided by `ding_event`.

File: ding_event.admin.inc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  $instance = field_read_instance('node', 'field_ding_event_price', 'ding_event');
+  if($instance !== FALSE) {
+    $instance['settings']['suffix'] = ' ' . $currency;
+    field_update_instance($instance);
+  }
```
Comment (kasperg) [inline ding_event.admin.inc]: Why is is done in a validation handler? If there is no good reason for it then I would expect such changes to be carried out in a submit handler.

File: ding_event.admin.inc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  $form['#validate'][] = 'ding_event_admin_currency_validate';
+
+  $form['ding_event_currency']['ding_event_currency_type'] = array(
+    '#type' => 'radios',
+    '#title' => t('Currency'),
+    '#options' => drupal_map_assoc(array('kr', '€')),
+    '#default_value' => $currency,
+  );
+
+// Save us the trouble of running array_filter.
```
Comment (kasperg) [inline ding_event.admin.inc]: Please correct indentation here.

File: ding_event.module [hunk 4]
```diff
@@ -6,6 +6,23 @@
 
 include_once 'ding_event.features.inc';
 
+
+/**
+ * Implements hook_menu
+ * Setting to administer currency shown on events
+ */
+function ding_event_menu() {
+  $items['admin/config/ting/currency'] = array(
```
Comment (kasperg) [inline ding_event.module]: This should go under `ding_events` configuration as the configuration form may be updated to contain additional settings related to events down the line:

This means:

- Path should be `admin/config/ding/events`
- Title should be adjusted accordingly
- Permissions should be adjusted accordingly
- Form method should be changed accordingly

File: ding_event.admin.inc (hunk 3 above)
Comment (pjohans) [inline ding_event.admin.inc]: ok

File: ding_event.module (hunk 4 above)
Comment (pjohans) [inline ding_event.module]: right again

File: ding_event.admin.inc (hunk 2 above)
Comment (pjohans) [inline ding_event.admin.inc]: okay

File: ding_event.admin.inc (hunk 1 above)
Comment (pjohans) [inline ding_event.admin.inc]: absolutely

Comment (pjohans) [inline ding_event.admin.inc]: i have moved the validation to ding_place2book. see 
https://github.com/ding2/ding_place2book/pull/7

File: ding_event.module [hunk 5]
```diff
@@ -6,6 +6,34 @@
...
+    'file' => 'system.admin.inc',
+    'file path' => drupal_get_path('module', 'system'),
+  );
+
+  $items['admin/config/ding_event/currency'] = array(
+    'title' => 'Event currency',
+    'description' => 'Select which currency to show on events.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_event_admin_currency_setting_form'),
+    'access arguments' => array('administer ting settings'),
```
Comment (kasperg) [inline ding_event.module]: I still think that this is an inappropriate permission to require for this page.

Ting settings are for managing the integration with OpenSearch - this is for handling the events. This should be a separate permission just as [ding_frontpage](https://github.com/ding2/ding_frontpage/blob/master/ding_frontpage.module#L9) has - preferably prefixed with `ding`.

**Comment by kasperg** [comment]
Reviewed and approved. Thanks!

**Pull request merged**